src/main/java/com/revature/security/extra/controllers/AuthenticationRestController.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package com.revature.security.extra.controllers;
+
+import com.revature.security.JwtAuthenticationRequest;
+import com.revature.security.extra.JwtTokenUtil;
+import com.revature.security.extra.JwtUser;
+import com.revature.revatureSecurity.JwtAuthenticationResponse;
+import javax.servlet.http.HttpServletRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.mobile.device.Device;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.security.core.userdetails.UserDetails;
+
+/**
+ *
+ * @author FayeRedd
+ */
+
+@RestController
+public class AuthenticationRestController {
+    
+    //This value can be found in the application.yml file
+    @Value("${jwt.header}")
+    private String tokenHeader;
+    
+    @Autowired
+    private AuthenticationManager authMan;
+    
+    //This is the token generator/validator. Can be replaced with future microservice providing/varifying tokens
+    @Autowired
+    private JwtTokenUtil tokenUtil;
+    
+    @Autowired
+    private UserDetailsService userDetServ;
+    
+    //This value can be found in the application.yml file
+    @RequestMapping(value = "${jwt.route.authentication.path}", method = RequestMethod.POST)
+    public ResponseEntity<?> createAuthenticationToken(@RequestBody JwtAuthenticationRequest authReq, Device dev) throws AuthenticationException{
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove usage of generic wildcard type. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS1452)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.springframework.security.core.AuthenticationException' which is a runtime exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)